AWSIoT/AWSIoTDataManager.m

```diff
@@ -333,6 +339,29 @@ - (void)enableMetricsCollection:(BOOL)enabled {
     [self.mqttClient setIsMetricsEnabled:enabled];
 }
 
+- (void)addUserMetaData:(NSDictionary<NSString *, NSString *> *)userMetaData {
+
+    // validate the length of username field
+    NSMutableString *userMetadata = [NSMutableString stringWithFormat:@"%@%@", @"?SDK=iOS&Version=", SDK_VERSION];
+    NSUInteger baseLength = [userMetadata length];
+
+    // Append each of the user-specified key-value pair to the connection username
+    if (userMetaData != [ NSNull null ]) {
+        for (id key in userMetaData) {
+            [userMetadata appendFormat:@"&%@=%@", key, [userMetaData objectForKey:key]];
+        }
+    }
+
+    if ([userMetadata length] > 255) {
+        AWSDDLogWarn(@"Total number of characters in username fields cannot exceed (%lu)", (255 - baseLength));
+        NSRange range = {0, MIN([userMetadata length], 255)};
```

Comment:
Can safely use [userMetaData substringToIndex:255] here as this is within the condition to simplify this further.